src/Compilers/Core/Desktop/CommandLine/CommonCommandLineArguments.cs

```diff
@@ -328,13 +328,51 @@ public IEnumerable<AnalyzerReference> ResolveAnalyzerReferences(IAnalyzerAssembl
         {
             foreach (CommandLineAnalyzerReference cmdLineReference in AnalyzerReferences)
             {
-                yield return ResolveAnalyzerReference(cmdLineReference, analyzerLoader)
-                    ?? (AnalyzerReference)new UnresolvedAnalyzerReference(cmdLineReference.FilePath);
+                string resolvedPath = ResolveAnalyzerReferencePath(cmdLineReference);
+
+                if (resolvedPath == null)
+                {
+                    yield return new UnresolvedAnalyzerReference(cmdLineReference.FilePath);
+                }
+
+                Assembly assembly = null;
+                try
+                {
+                    assembly = analyzerLoader.LoadFromPath(resolvedPath);
+                }
+                catch (Exception)
+                {
```

Comment:
Yes, Jared is right. This will cause us to not report any assembly load failures diagnostics, which the current implementation does: http://source.roslyn.io/#Microsoft.CodeAnalysis.Desktop/AnalyzerFileReference.cs,219. We need a unit test for this scenario.